packages/framework/src/Form/Admin/Administrator/AdministratorFormType.php

```diff
@@ -160,6 +160,35 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 'back_route' => 'admin_administrator_list',
                 'entity' => $adminToEdit,
             ]);
+
+        if ($options['scenario'] !== self::SCENARIO_EDIT || !$this->canWorkWithRoles($adminToEdit) || !$this->getCurrentAdministrator()->isSuperadmin()) {
+            return;
+        }
+
+        $builderPromoteGroup = $builder->create('promote', GroupType::class, [
+            'label' => 'Super Administrator',
+        ]);
+
+        $builderPromoteGroup->add('promoteToSuperadmin', DisplayOnlyUrlType::class, [
+            'route' => 'admin_administrator_promote-to-superadmin',
+            'route_params' => [
+                'id' => $adminToEdit->getId(),
+                RouteCsrfProtector::CSRF_TOKEN_REQUEST_PARAMETER => $this->routeCsrfProtector->getCsrfTokenByRoute(
+                    'admin_administrator_promote-to-superadmin',
+                ),
+            ],
+            'route_attr' => [
+                'data-confirm-window' => null,
+                'data-confirm-style' => 'warning',
+                'data-confirm-message' => t('Are you sure you want to promote this user to Superadmin?'),
+            ],
+            'label' => 'Superadmin permissions',
+            'help' => 'Superadmins have all permissions including changing system configuration and access to hidden administration sections.',
+            'route_label' => t('Promote user to Superadmin'),
+            'link_target' => '_self',
+        ]);
+
+        $builder->add($builderPromoteGroup);
```

Comment:
This looks cool, but why don't you just create that action as a grid action? It can show all that information in that confirmation window.

Also, this is just another "Role Group", but in UI it's separated, and it looks like this is something different. 

I understand why you implemented it this way, but isn't that overkill?

Reply:
I didn't want to put it in the grid to avoid having such a niche scenario available from the list. 
I see "promote to superadmin" as something that should be carefully thought through, and the separation helps understand that it's not "just another role", but something really dangerous.